Source/FolioReaderCenter.swift

```diff
@@ -335,259 +357,267 @@ open class FolioReaderCenter: UIViewController, UICollectionViewDelegate, UIColl
 
     func hideBars() {
 
-        if readerConfig.shouldHideNavigationOnTap == false { return }
+        guard (self.readerConfig.shouldHideNavigationOnTap == true) else {
+            return
+        }
 
-        let shouldHide = true
-        FolioReader.shared.readerContainer.shouldHideStatusBar = shouldHide
-        
-        UIView.animate(withDuration: 0.25, animations: {
-            FolioReader.shared.readerContainer.setNeedsStatusBarAppearanceUpdate()
-            
-            // Show minutes indicator
-//            self.pageIndicatorView.minutesLabel.alpha = 0
-        })
-        navigationController?.setNavigationBarHidden(shouldHide, animated: true)
+        self.updateBarsStatus(true)
     }
-    
+
     func showBars() {
-        configureNavBar()
-        
-        let shouldHide = false
-        FolioReader.shared.readerContainer.shouldHideStatusBar = shouldHide
-        
-        UIView.animate(withDuration: 0.25, animations: {
-            FolioReader.shared.readerContainer.setNeedsStatusBarAppearanceUpdate()
-        })
-        navigationController?.setNavigationBarHidden(shouldHide, animated: true)
+        self.configureNavBar()
+        self.updateBarsStatus(false)
     }
-    
+
     func toggleBars() {
```

Comment:
There is a bug on bar toggle, on the Sherlock Holmes book, tap to show is not showing the bar, it toggle fast but keeps hidden.